src/components/upgrader/huddlygoUpgrader.ts

```diff
@@ -45,12 +47,23 @@ export default class HuddlyGoUpgrader extends EventEmitter implements IDeviceUpg
     }
   }
 
+  emitProgressStatus(statusString?: string) {
+    if (statusString) this._upgradeStatus.statusString = statusString;
+    this.emit(CameraEvents.UPGRADE_PROGRESS, this._upgradeStatus.getStatus());
+  }
+
   async start(): Promise<void> {
+    const step = new UpgradeStatusStep('Upgrading camera');
+    this._upgradeStatus = new UpgradeStatus([step]);
+    this.emitProgressStatus('Starting upgrade');
     this.emit(CameraEvents.UPGRADE_START);
-    this.doUpgrade();
+    step.progress = 1;
+    this.emitProgressStatus();
```

Comment:
I find this a bit cryptic, if you don't provide a string it will automatically the status from ```this._upgradeStatus```